### ISSUE TITLE
RNN `dropout` arg needs better documentation

### ISSUE THREAD

**Issue by flauted** [pytorch/pytorch#5985]
The documentation for the argument ``dropout`` of the LSTM module [here](http://pytorch.org/docs/master/nn.html#torch.nn.LSTM) confused me. Here's the quote:

>If non-zero, introduces a dropout layer on the outputs of each RNN layer except the last layer

GRU uses the same documentation. Initially I assumed that meant supplying a value other than `False` will implement a dropout layer with a default dropout probability. 

Today I dug through the source code. I went through LSTM, RNN base, Module, THNN backends, Function backends, functions RNN, functions Autograd RNN, and finally [functions Stacked RNN](https://github.com/pytorch/pytorch/blob/master/torch/nn/_functions/rnn.py#L68) before I figured out that the argument is a float probability and not a boolean.

I did some searches on the forums. I'm not the first person to become confused (Exhibit [A](https://discuss.pytorch.org/t/lstm-dropout-clarification-of-last-layer/5588), [B](https://discuss.pytorch.org/t/dropout-in-lstm/7784), [C](https://discuss.pytorch.org/t/dropout-for-rnns/633)). It wasn't until I read through A that I figured out by "layer" it didn't mean the last unwrapped layer but rather the last stacked layer. That is to say, using the arg when `num_layers` is 1 actually did nothing. Possibly worthy of a warning...

I opened a topic on the forums [here](https://discuss.pytorch.org/t/does-rnn-dropout-arg-need-better-documentation/15407). Didn't gain traction, but the one reply at this time suggested an issue or PR.

**Tl;dr: The docs for `nn.LSTM` and `nn.GRU` should say the arg is a float in [0, 1], should clarify what's meant by layer, and maybe the source should issue a warning when `dropout` is not 0 and `num_layers` is 1.**
